src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -133,24 +133,46 @@ private AdapterBinding getUserAdapterBinding(Property property, JsonbContext jso
 
     private PropertyCustomization introspectCustomization(Property property, JsonbContext jsonbContext) {
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
-        final CustomizationBuilder builder = new CustomizationBuilder();
+        final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
         if (introspector.isTransient(property)) {
             builder.setJsonbTransient(true);
             return builder.buildPropertyCustomization();
         }
         builder.setJsonReadName(introspector.getJsonbPropertyJsonReadName(property));
         builder.setJsonWriteName(introspector.getJsonbPropertyJsonWriteName(property));
-        builder.setNillable(classModel.getClassCustomization().isNillable()
-                || introspector.isPropertyNillable(property));
+        builder.setNillable(classModel.getClassCustomization().isNillable() || introspector.isPropertyNillable(property));
         builder.setAdapterInfo(getUserAdapterBinding(property, jsonbContext));
         builder.setSerializerBinding(getUserSerializerBinding(property, jsonbContext));
         builder.setDeserializerBinding(introspector.getDeserializerBinding(property));
         builder.setDateFormatter(introspector.getJsonbDateFormat(property));
-        builder.setNumberFormat(introspector.getJsonbNumberFormat(property));
+        introspectNumberFormatter(property, introspector, builder);
         return builder.buildPropertyCustomization();
     }
 
+    private void introspectNumberFormatter(Property property, AnnotationIntrospector introspector, PropertyCustomizationBuilder builder) {
+        /*
+         * If @JsonbNumberFormat is placed on getter implementation must use this format on serialization.
+         * If @JsonbNumberFormat is placed on setter implementation must use this format on deserialization.
+         * If @JsonbNumberFormat is placed on field implementation must use this format on serialization and deserialization.
+         */
+        Map<AnnotationTarget, JsonbNumberFormatter> jsonNumberFormatCategorized = introspector.getJsonNumberFormatter(property);
+        if(jsonNumberFormatCategorized.keySet().contains(AnnotationTarget.PROPERTY)) {
```

Comment:
I got confused. I was thinking just the opposite way. I think I misinterpreted the issue description. Should I change it to the way in which number formatter would be evaluated in following order?

1. if there is any number formatter for getter/setter they would be used
2. if there is no formatter for getter/setter and there is one for field, the field level formatter would be used
3. if there is no formatter for getter, setter or field and there is a formatter on class level, the class level formatter would be used

So the priority from high to low would be : getter/setter -> field -> class